rfd/0163-vnet.md

```diff
@@ -0,0 +1,813 @@
+API](https://forums.developer.apple.com/forums/thread/701118?answerId=705661022#705661022), it gets
+us past the problem of acquiring root privileges and lets us focus on validating VNet as a tool.
+In the [Alternatives to osascript](#alternatives-to-osascript) section, we describe other approaches
+which offer better UX at the cost of development time.
+
+#### Guiding the user towards VNet
+
+Before VNet, the user would access web and TCP apps through the unified resources view in Connect:
+
+<img width="871" alt="Unified Resources view in Connect v15.0.0" src="assets/0163-apps-in-connect-v15.png">
+
+Clicking "Launch" next to a web app would launch this app in the browser through the `/web/launch`
+URL of the proxy service. The user could also set up a local proxy for the specific app by clicking
+on the three dots menu and choosing "Set up connection". For TCP apps, clicking "Connect" would set
+up a local proxy for that app.
+
+This changes with the introduction of VNet. Here's a screenshot from Teleport Connect in cluster
+`teleport-local.dev` running VNet:
+
+<img width="871" alt="Unified Resources view in Connect with VNet turned on" src="assets/0163-apps-with-vnet.png">
+
+* dumper is a web app without `spec.vnet_addr`. "Launch" opens the app in the browser through the
+  `/web/launch` of the proxy service. The user can select "Launch through VNet" from the three dots
+  menu to open `http://dumper.teleport-local` or "Copy VNet address" to copy that address. The
+  address in the card is `https://dumper.teleport-local.dev` as usual.
+* httpbin is a web app with `spec.vnet_addr` set to `httpbin.company.private` and `.company.private`
+  is configured as the suffix of a DNS zone in the `vnet` resource. "Launch" starts VNet if it's not
+  already running and opens the VNet address of the app. The user can select "Launch through Web UI"
+  from the three dots menu. The address in the card is `http://httpbin.company.private`.
+* tcp-postgres is a TCP app without `spec.vnet_addr`. If VNet is not running, the button says
+  "Connect" and it starts VNet, copies the VNet address of the app to the clipboard and shows a
+  notification in the bottom right saying "Copied `tcp-postgres.teleport-local.dev.internal` to the
+  clipboard.". If VNet is running, the button simply says "Copy". The user can select "Set up a
+  legacy connection" which starts the equivalent of `tsh proxy app tcp-postgres` that's available
+  today.
+
+Setting up legacy connections is kept for users who might not be able to use VNet, either because
+they lack root privileges or VNet would conflict with other software running on the computer.
+
+In case of web apps, we don't know if they're intended to be used through VNet or not. The only hint
+we can get is the existence of `spec.vnet_addr` which indicates that the given web app is, for
+example, an HTTP API that users might want to connect to through VNet.
```

Comment:
I don't think the difference matters much for web apps, we should treat them the same. If it's an HTTP API they shouldn't want to open it in a browser anyway. I can't think of any good reason to open a VNet address in a browser. The only difference if `vnet_addr` is set should be the result of "Copy VNet address"